zerver/lib/actions.py

```diff
@@ -3927,39 +3930,36 @@ def do_update_user_presence(user_profile: UserProfile,
                             status: int) -> None:
     client = consolidate_client(client)
 
+    # TODO: While we probably DO want creating an account to
+    # automatically create a first `UserPresence` object with
+    # last_connected_time and last_active_time as the current time,
+    # our presence tests don't understand this, and it'd be perhaps
+    # wrong for some cases of account creation via the API.  So we may
+    # want a "never" value here as the default.
```

Comment:
We probably don't have to fix this TODO before merging.

Reply:
This actually creates these default values on the first presence update - which does not occur on user creation :thinking: I think that actually works well, though requires keeping it in mind when writing tests - that the first presence update the test sends will actually create a UserPresence with both active_time and connected_time set to now. Unless the test creates some initial data to avoid invoking that behavior. But it's only a handful of tests where this comes into play, so I think we can be okay with this?